This PR fixes three critical issues preventing fax functionality from working correctly: missing job parameters causing 500 errors, unencrypted password storage, and UI/formatting issues in the fax cover page.

D3V41 opened the pull request:
## Issues Fixed

  ### 1. Missing Fax Job Parameters (500 Error)
  **Problem**: Fax queue action was passing an empty HashMap to `faxManager.createAndSaveFaxJob()`, causing 500 errors when attempting to send faxes.

  **Solution**:
  - Properly collect all fax job parameters from request
  - Pass complete parameter map including: faxFilePath, recipient, recipientFaxNumber, senderFaxNumber, demographicNo, comments, coverpage, copyToRecipients
  - Added missing getter/setter methods in `Fax2Action.java`

  ---

  ### 2. Missing Password Encryption/Decryption
  **Problem**: Missing Fax password Encryption/Decryption logic.

  **Solution**:
  - Added missing code from the OpenO


  ---

  ### 3. Fax Send Button and Recipient Format Issues
  **Problem**: Multiple issues preventing successful fax sending:
  - Security validation errors when getting PDF page count
  - Incorrect recipient JSON format breaking server-side parsing
  - Preview panel showing stale data after submission
  - Recipient list displaying incorrectly after fax sent

  **Solution**:
  - **File path resolution**: Resolve to full `OscarDocument` path before PDF operations in `FaxManagerImpl.java`
  - **JSON format fix**: Changed recipient format from `'name':'value'` to proper JSON `"name":"value"`
  - **UI improvements**:
    - Hide preview panel after submission using `${ empty faxSuccessful }` check
    - Only show recipient list before submission
    - Escape double quotes in JavaScript to prevent JSON breaking

Closes #887

## Summary by Sourcery

Fix fax functionality by properly collecting job parameters, securing password storage, and resolving UI/formatting issues in the fax cover page.

Bug Fixes:
- Collect and pass complete fax job parameters to faxManager to prevent 500 errors
- Implement encryption/decryption for stored fax passwords in FaxConfig
- Correct recipient JSON formatting, escape quotes, and toggle preview panel visibility to fix UI issues

Enhancements:
- Resolve file paths to full OscarDocument before PDF page count to avoid security validation errors

D3V41 commented on src/main/webapp/fax/CoverPage.jsp:
```diff
@@ -317,26 +317,29 @@
+							  			<div class="row">
+								  			<div class="col-sm-12 input-group recipientGroup">
+								  				<label></label>
+											      <input type="text" class="form-control" value="${ recipient.name } ${ recipient.fax }" disabled/>
```
Resolved

D3V41 commented on src/main/java/ca/openosp/openo/fax/action/Fax2Action.java:
```diff
@@ -124,7 +124,20 @@ public String queue() {
+                .comments(comments)
+                .coverpage(coverpage)
+                .copyToRecipients(copyToRecipients)
+                .build();
```
Resolved

yingbull commented:
Hi Deval, can you add the path sanitation in? I believe it should be straight forward; IIRC, @LiamStanziani or @sebastian-j-ibanez did this recently through an apache library.

D3V41 commented:
Hi Michael, I've added the path sanitization. I used Apache Commons IO's `FilenameUtils.getName()` to strip any path components from filenames. Ans also added canonical path resolution to prevent symlink-based traversal attacks.

The main changes are in `NioFileManagerImpl.copyFileToOscarDocuments()` where I replaced the manual path handling with the Apache library, and in `FaxManagerImpl.createFaxJob()` where I'm using the `resolveAndValidateFilePath()` method. `resolveAndValidateFilePath()` method is more robust than just using Apache's FilenameUtils or FileUtils alone, as it combines multiple validation layers. Note that the path validation is already handled in the NioFileManager class.
